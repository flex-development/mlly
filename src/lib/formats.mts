/**
 * @file formats
 * @module mlly/lib/formats
 */

import type { ModuleFormat } from '@flex-development/mlly'

/**
 * Module formats.
 *
 * @see {@linkcode ModuleFormat}
 *
 * @enum {ModuleFormat}
 */
enum formats {
  builtin = 'builtin',
  commonjs = 'commonjs',
  cts = 'commonjs-typescript',
  json = 'json',
  module = 'module',
  mts = 'module-typescript',
  wasm = 'wasm'
}

export default formats
