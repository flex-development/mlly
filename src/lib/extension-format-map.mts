/**
 * @file extensionFormatMap
 * @module mlly/lib/extensionFormatMap
 */

import formats from '#lib/formats'
import type { ModuleFormat } from '@flex-development/mlly'
import type { Ext } from '@flex-development/pathe'

/**
 * File extension to module format map.
 *
 * @see {@linkcode Ext}
 * @see {@linkcode ModuleFormat}
 *
 * @const {Map<Ext, ModuleFormat>} extensionFormatMap
 */
const extensionFormatMap: Map<Ext, ModuleFormat> = new Map<Ext, ModuleFormat>([
  ['.cjs', formats.commonjs],
  ['.cts', formats.cts],
  ['.d.cts', formats.cts],
  ['.d.mts', formats.mts],
  ['.d.ts', formats.mts],
  ['.js', formats.module],
  ['.json', formats.json],
  ['.jsx', formats.module],
  ['.mjs', formats.module],
  ['.mts', formats.mts],
  ['.node', formats.commonjs],
  ['.ts', formats.mts],
  ['.tsx', formats.mts],
  ['.wasm', formats.wasm]
])

export default extensionFormatMap
