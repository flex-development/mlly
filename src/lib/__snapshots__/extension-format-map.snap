// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:lib/extensionFormatMap > should map extensions to module formats 1`] = `
Map {
  ".cjs" => "commonjs",
  ".cts" => "commonjs-typescript",
  ".d.cts" => "commonjs-typescript",
  ".d.mts" => "module-typescript",
  ".d.ts" => "module-typescript",
  ".js" => "module",
  ".json" => "json",
  ".jsx" => "module",
  ".mjs" => "module",
  ".mts" => "module-typescript",
  ".node" => "commonjs",
  ".ts" => "module-typescript",
  ".tsx" => "module-typescript",
  ".wasm" => "wasm",
}
`;
